kernel/posix/pthread_semaphore.c

```diff
@@ -0,0 +1,111 @@
+/*
+ * Copyright (c) 2018 Intel Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <pthread.h>
+
+int errno;
+
+int sem_destroy(sem_t *semaphore)
+{
+
+	if (k_sem_count_get(semaphore->sem) != 0) {
+		errno = EINVAL;
+		return -1;
+	}
+
+	k_sem_reset(semaphore->sem);
+
+	return 0;
+}
+
+int sem_getvalue(sem_t *semaphore, int *value)
+{
+	if (semaphore == NULL) {
```

Comment:
Please be consistent with the null checks: either do all of them, or do none of them. Consult with the POSIX documentation to check if the behavior is defined or not when a NULL pointer is passed. This is sometimes up for interpretation (e.g. the standard might say that the pointer must point to a valid object, and NULL isn't a valid object).